connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/PluginUtilsTest.java

```diff
@@ -490,6 +504,168 @@ public void testPluginUrlsWithRelativeSymlinkForwards() throws Exception {
+        assertEquals(MockSinkConnector.class.getName(), aliases.get("MockSink"));
+        assertEquals(MockSourceConnector.class.getName(), aliases.get("MockSourceConnector"));
+        assertEquals(MockSourceConnector.class.getName(), aliases.get("MockSource"));
+        assertEquals(CollidingConverter.class.getName(), aliases.get("CollidingConverter"));
+        assertEquals(CollidingConverter.class.getName(), aliases.get("Colliding"));
+    }
+
+
+    @Test
+    public void testMultiVersionAlias() {
+        SortedSet<PluginDesc<SinkConnector>> sinkConnectors = new TreeSet<>();
+        // distinct versions don't cause an alias collision (the class name is the same)
+        sinkConnectors.add(new PluginDesc<>(MockSinkConnector.class, null, MockSinkConnector.class.getClassLoader()));
+        sinkConnectors.add(new PluginDesc<>(MockSinkConnector.class, "1.0", MockSinkConnector.class.getClassLoader()));
+        assertEquals(2, sinkConnectors.size());
+        PluginScanResult result = new PluginScanResult(
+                sinkConnectors,
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet()
+        );
+        Map<String, String> aliases = PluginUtils.computeAliases(result);
+        assertEquals(MockSinkConnector.class.getName(), aliases.get("MockSinkConnector"));
+        assertEquals(MockSinkConnector.class.getName(), aliases.get("MockSink"));
+    }
+
+    @Test
+    public void testCollidingPrunedAlias() {
+        SortedSet<PluginDesc<Converter>> converters = new TreeSet<>();
+        converters.add(new PluginDesc<>(CollidingConverter.class, null, CollidingConverter.class.getClassLoader()));
+        SortedSet<PluginDesc<HeaderConverter>> headerConverters = new TreeSet<>();
+        headerConverters.add(new PluginDesc<>(CollidingHeaderConverter.class, null, CollidingHeaderConverter.class.getClassLoader()));
+        PluginScanResult result = new PluginScanResult(
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                converters,
+                headerConverters,
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet()
+        );
+        Map<String, String> aliases = PluginUtils.computeAliases(result);
+        assertEquals(CollidingConverter.class.getName(), aliases.get("CollidingConverter"));
+        assertEquals(CollidingHeaderConverter.class.getName(), aliases.get("CollidingHeaderConverter"));
+        assertFalse(aliases.containsKey("Colliding"));
```

Comment:
Same thought RE comparing whole `Map` objects:

```suggestion
        Map<String, String> actualAliases = PluginUtils.computeAliases(result);

        Map<String, String> expectedAliases = new HashMap<>();
        expectedAliases.put("CollidingConverter", CollidingConverter.class.getName());
        expectedAliases.put("CollidingHeaderConverter", CollidingHeaderConverter.class.getName());
        assertEquals(expectedAliases, actualAliases);
```